### ISSUE TITLE
Maintain a Working Example of Canvas Source Or...

### ISSUE THREAD

**Issue by lvanngo** [mapbox/mapbox-gl-js#9149]
**mapbox-gl-js version**:

### Question

There is no working example for the canvas source type that is compatible with the current version of mapbox-gl js.  The docs do provide some information but trying to hook it up to current version does not produce working result.  All other source types seem to work which makes me doubtful whether it is a current bug.  There does not seem to be any elegant alternative to the canvas source type that I can see--Are we suppose to use it? 

### Links to related documentation
https://docs.mapbox.com/mapbox-gl-js/api/#canvassource

**Comment by wlfei0502** [comment]
You can refer to  [the example](https://docs.mapbox.com/mapbox-gl-js/example/image-on-a-map/) of `ImageSource`, because `CanvasSource` extends `ImageSource`.

**Comment by lvanngo** [comment]
Thank you for the tips.  I did realize this but looked at it again.  I tried using addSource on top of the image example and it works!  I tried this with the canvas source by keeping all the code except changing the source to canvas source and it did not work.  I finally realized why--I think it makes a difference where we add the canvas element of the source.  When I moved it from inside my framework web component to the body directly, it works!  Thanks again :)
